README.md

```diff
@@ -7,43 +7,40 @@
 [![dev chat](https://discordapp.com/api/guilds/299006062323826688/widget.png?style=shield)](https://discord.gg/ga2xZXk)
 
 
-The source code of `karaoke!` ruleset, running on [osu!lazer](https://github.com/ppy/osu).
```

Comment:
This repo is based on lazer(and osu-framework), but might only run on osu!

Reply:
What do you mean by might? Wouldn’t it only run on osu lazer?

Reply:
Oops, might `not` only run on osu!lazer, my bad.
Lazer is MIT license, so anyone can use the code as they like.

Reply:
So if anyone modified their game client base on [osu!lazer's package](https://www.nuget.org/packages/ppy.osu.Game/), karaoke ruleset can still run on it.

Reply:
Maybe it's better to say it's `based on`.

Reply:
Ahh, makes more sense. I will make it clearer

Reply:
Fixed

Reply:
`based off of`?

Reply:
Late to the discussion, but "running on" is more correct than "based on".

Reply:
It's ok for me to any of them, as i'm not native speaker🤔.

Reply:
Yeah that’s what I figured initially since it was only running on lazer at the moment, but Andy phrased it so that I thought it might eventually run on another project made from the osu framework so in that case I figured we could say based off of as it fits and wont need to be changed in the future if that ever occurs

Reply:
It is purely preference though, I think running on fits better but based off of/on still works

Reply:
Hmm...
I'm just doesn't know `based off of` means.
If this word (`The source code of the karaoke! ruleset, based off of osu!lazer`) has no grammar issue, then it's ok.

Reply:
Just guessing based off your username but if you know Chinese it is the same as “基于”, it is a different way to say “based on”. I will change it to be “running on” because I agree with bloom it makes more sense

Reply:
Did you know Chinese?
Cool